src/openrct2/world/Sprite.cpp

```diff
@@ -323,9 +308,11 @@ rct_sprite_checksum sprite_checksum()
         }
 
         _spriteHashAlg->Clear();
+        OpenRCT2::MemoryStream ms;
+        DataSerialiser ds(true, ms);
 
-        ComputeChecksumForEntityTypes<Guest, Staff, Vehicle, Litter>(_spriteHashAlg.get());
-
+        NetworkSerialiseEntityTypes<Guest, Staff, Vehicle, Litter>(ds);
+        _spriteHashAlg->Update(ms.GetData(), ms.GetLength());
```

Comment:
This introduces a lot of overhead, this is almost as expensive as saving the park now minus the file i/o. 
1. MemoryStream uses dynamic memory allocation
2. All data is copied first into the buffer.
3. Has to run over the copied data again to compute the checksum

Theres probably a trick behind this by introducing a new stream that inherhits from IStream, lets call it ChecksumStream.

The serializer will feed all data via write so the checksum can be done on the fly.